crates/sui-core/src/transaction_manager.rs

```diff
@@ -83,6 +83,15 @@ impl TransactionManager {
         let inner = &mut self.inner.lock();
         for cert in certs {
             let digest = *cert.digest();
+
+            if epoch_store
```

Comment:
I think we can move this to `AuthorityState::enqueue_certificates_for_execution()` and call the function from Narwhal, so we can avoid applying this logic to checkpoint executions.